### ISSUE TITLE
Watchout about x86-64 branch server physics. It is somewhat broken.

### ISSUE THREAD

**Issue by Grocel** [dvdvideo1234/TrackAssemblyTool#60]
Server running the x86-64 branch have been proven to have broken physics. I think this quite important for you and your TA Tool to know about. The physics issue especially affect trains and tracks in all sorts of variants. Even the best adjusted slider track combos will derail after a short ride, even at slow speeds. For more details please see my report in the GMod repository:

[Facepunch/garrysmod-issues#5968 - Unstable collision detection in physics on x86-64 branch](https://github.com/Facepunch/garrysmod-issues/issues/5968)

At first the issue had been reported to SligWolf and me there:
[SW Addons - Bug Reports | Problem with the trains on x64 Chromium Beta](https://steamcommunity.com/groups/SligWolfAddons/discussions/1/4513255384647889127/)

Then we investigated it further more along with other experienced developers and confirmed the issue existing on all platforms (CPU/OS) of x86-64 branch servers (Dedicated server, single player or listen servers).

**Comment by dvdvideo1234** [comment]
Yes I heard. xD Sadly I am in @SligWolf's shoes in this one 😉. He and you explain the problem quite accurately too. 😀

I wonder if this can be solved with bumping the physics iterations. 🤔🤔

**Comment by dvdvideo1234** [comment]
I see this was not the case however 4 months ago:
https://m.youtube.com/watch?v=guW5F27dhLE

I never liked gmod physical collisions for trains so I did the dedicated Lua maglevs and E2 ones:
https://m.youtube.com/watch?v=G_xcm1OFVV0&list=PLtUiNPRuZQ2lH85uGrUN1aaYPSUymzpKv&index=16&pp=gAQBiAQB

**Comment by Grocel** [comment]
> Yes I heard. xD Sadly I am in @SligWolf's shoes in this one 😉. He and you explain the problem quite accurately too. 😀
> 
> I wonder if this can be solved with bumping the physics iterations. 🤔🤔

This has been ruled out. Changing iteration and tick settings does nothing on this issue.

> I see this was not the case however 4 months ago: https://m.youtube.com/watch?v=guW5F27dhLE
> 
> I never liked gmod physical collision for trains so I did the dedicated Lua maglevs and E2 ones: https://m.youtube.com/watch?v=G_xcm1OFVV0&list=PLtUiNPRuZQ2lH85uGrUN1aaYPSUymzpKv&index=16&pp=gAQBiAQB

Remember the issue only happens if the server/listenserver runs on branch `x86-64`. On the `main` branch (`dev` and `prerelease` too) is still not affected, in case your have missed that detail. They are still fine, for now.

I don't think your videos were recorded on `x86-64` neither. The `x86-64` issue is at least a year old, or maybe even longer. The  issue shows the physics being different between `main` and `x86-64`.

Lua Maglevs was also an idea that came up multiple times, but Lua tracers running from many trains/wagons are a way more expensive than the C++ physics code running. It is not feasible for every use case neither. For Sligwolf and me it would be a way to difficult to code properly too. Sligwolf and me never needed such workarounds. I am still impressed, though.

Everything you need to know about the physics problem is summarized in my linked GMod issue. It is very unlikely that it can be fixed just with changed models and "optimized" sliders/wheels alone, especially without heavy drawbacks.

It is a problem that affects everyone in the train building community or even other GMod players too. That's why I insist that people know about this, get the details straight and don't be confused about them.

**Comment by dvdvideo1234** [comment]
Just merged the control statements PR. It will help more or less with this. We will see...
https://github.com/dvdvideo1234/TrackAssemblyTool/issues/60

**Comment by Grocel** [comment]
@dvdvideo1234 Are you sure the PR #59 is relevant to this? Also you linked to #60 which is this issue thread.

**Comment by dvdvideo1234** [comment]
I doubt it will change anything about this issue as I did not change anything [in the spawn function](https://github.com/dvdvideo1234/TrackAssemblyTool/pull/59/files#diff-1504cf3838e6c1a40de88518ce42b3aaa7a6d9f9afbdf1d58c447f9796d33e51L4658). I'll just pin it. As it is a Gmod issue we will just see as it is too early to tell. Maybe Gmod devs will fix it ;). I wonder if [Jolt](https://github.com/misyltoad/VPhysics-Jolt) will have any effect on this.

**Comment by Grocel** [comment]
It seems like that non x86-64 version is affected too if you test it on a dedicated server. It was reported to me just recently. The guy (@MartyX5555) who reported it believes that the issue could be related to changes that were made in PhysicsInitConvex some time ago.

**Comment by Grocel** [comment]
It has been confirmed by @MartyX5555 that the issue does not exist on non x86-64 versions of the dedicated server. The new report had a mistake with the version tested. The report has been changed accordingly.

**Comment by Grocel** [comment]
Hi @dvdvideo1234,

@SligWolf had been cooking this week and made a proper reproduction setup on this issue.
https://github.com/Facepunch/garrysmod-issues/issues/5968#issuecomment-2719058065

With Videos, a test map running on most source engine games and map source files.

**Comment by dvdvideo1234** [comment]
Time for Valve to do this fix @Grocel. They really took their time with it :D
https://github.com/ValveSoftware/Source-1-Games/issues/7426#issuecomment-3132558351
